test_suite/tests/json.rs

```diff
@@ -197,6 +196,30 @@ fn test_struct() {
     }));
 }
 
+#[test]
+fn test_struct_with_phantom() {
+    use scale_info::prelude::marker::PhantomData;
+    #[derive(TypeInfo)]
+    struct Struct<T> {
+        a: i32,
+        b: PhantomData<T>,
+    }
+
+    assert_json_for_type::<Struct<u8>>(json!({
+        "path": ["json", "Struct"],
+        "params": [1],
+        "def": {
+            "composite": {
+                "fields": [
+                    // TODO: dp – I was expecting "a" to be type 1 and "b" to be type 2; what am I missing?
+                    { "name": "a", "type": 2, "typeName": "i32" },
+                    { "name": "b", "type": 1, "typeName": "PhantomData<T>" },
```

Comment:
@ascjones I was a little surprised by the values for `"type"` here. Is it expected that the type parameter goes first into the registry, as type 1? Or am I missing something?

Reply:
Encoding is a bit complicated due to more exotic use cases such as recursive types.

Reply:
So not a source for concern then?

Reply:
As long as the encoding is deterministic everything is alright. Exact ordering does not matter as long as it is consistent.